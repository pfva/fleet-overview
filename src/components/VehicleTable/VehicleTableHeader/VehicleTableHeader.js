import VehicleHeaderCell from '../VehicleTableHeader/VehicleHeaderCell/VehicleHeaderCell';
import styles from './VehicleTableHeader.module.css';

const VehicleTableHeader = () => {
  return (
    <div className={styles.vehicleTableHeader}>
      <VehicleHeaderCell title={'id'} />
      <VehicleHeaderCell title={'name'} />
      <VehicleHeaderCell title={'driver'} />
      <VehicleHeaderCell title={'status'} />
      <VehicleHeaderCell title={'fuelType'} />
      <VehicleHeaderCell title={'equipments'} />
    </div>
  );
};

export default VehicleTableHeader;
