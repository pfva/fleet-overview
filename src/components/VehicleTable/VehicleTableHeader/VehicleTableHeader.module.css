.vehicleTableHeader {
  display: flex;
  padding: var(--spacing-normal);
  justify-content: space-between;
}

@media screen and (max-width: 425px) {
  .vehicleTableHeader {
    display: none;
  }
}
