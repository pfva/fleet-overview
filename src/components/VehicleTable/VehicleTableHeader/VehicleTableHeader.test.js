import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import VehicleTableHeader from './VehicleTableHeader';

describe('VehicleTableHeader', () => {
  it('renders without crashing', () => {
    const tree = renderer.create(<VehicleTableHeader />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the id column header', () => {
    render(<VehicleTableHeader />);
    const id = screen.getByText(/id/i);
    expect(id).toBeInTheDocument();
  });

  it('renders the name column header', () => {
    render(<VehicleTableHeader />);
    const name = screen.getByText(/name/i);
    expect(name).toBeInTheDocument();
  });

  it('renders the driver column header', () => {
    render(<VehicleTableHeader />);
    const driver = screen.getByText(/driver/i);
    expect(driver).toBeInTheDocument();
  });

  it('renders the status column header', () => {
    render(<VehicleTableHeader />);
    const status = screen.getByText(/status/i);
    expect(status).toBeInTheDocument();
  });

  it('renders the fuelType column header', () => {
    render(<VehicleTableHeader />);
    const fuelType = screen.getByText(/fuelType/i);
    expect(fuelType).toBeInTheDocument();
  });

  it('renders the equipments column header', () => {
    render(<VehicleTableHeader />);
    const equipments = screen.getByText(/equipments/i);
    expect(equipments).toBeInTheDocument();
  });
});
