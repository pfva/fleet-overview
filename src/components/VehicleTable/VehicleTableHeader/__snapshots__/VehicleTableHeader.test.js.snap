// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleTableHeader renders without crashing 1`] = `
<div
  className="vehicleTableHeader"
>
  <p
    className="vehicleHeaderCell"
  >
    id
  </p>
  <p
    className="vehicleHeaderCell"
  >
    name
  </p>
  <p
    className="vehicleHeaderCell"
  >
    driver
  </p>
  <p
    className="vehicleHeaderCell"
  >
    status
  </p>
  <p
    className="vehicleHeaderCell"
  >
    fuelType
  </p>
  <p
    className="vehicleHeaderCell"
  >
    equipments
  </p>
</div>
`;
